docs/howto/danceable-lyrics-input-tool/README.md

```diff
@@ -0,0 +1,485 @@
+    )
+    .filter(pl.col("explicit") == 0)
+    .collect()
+)
+
+track_artists = (
+    pl.scan_csv(self.DATASETS_BASE + "r_track_artist.csv")
+    .select(pl.col("track_id"), pl.col("artist_id"))
+    .sort("track_id")
+    .collect()
+)
+
+audio_features = (
+    pl.scan_csv(self.DATASETS_BASE + "audio_features.csv")
+    .select(pl.col("id").alias("track_id"), "danceability", "energy", "tempo",)
+    .filter(pl.col("danceability").is_between(*self.DANCEABILITY_RANGE))
+    .filter(pl.col("energy").is_between(*self.ENERGY_RANGE))
+    .filter(pl.col("tempo").is_between(*self.TEMPO_RANGE))
+    .sort("track_id")
+    .collect()
+)
+
+danceability_tracks = (
+    track_artists.join(artists, on="artist_id")
+    .join(tracks, on="track_id")
+    .join(audio_features, on="track_id")
+)
+```
+
+If we were to `print(danceability_tracks)` we'd see a header similar to the following:
+
+```txt
+track_id | artist_id | artist_name | track_name | explicit | danceability | energy | tempo
+```
+As you can see, this is a combination (joined) from multiple sources.
+
+
+Now that we have both a sample of songs, their lyrics containing our terms, and track information including "danceability", let's find our matches! Add the following code:
+
+```python
+self.provider.io.info(f"{self.name} calculating final results...")
+
+matches = pl.DataFrame()
+for row in sample.rows(named=True):
+    artist = row["artist_name"]
+    track = row["track_name"]
+    m = (
+        danceability_tracks.select(
+            "artist_name",
+            "track_name",
+            "danceability",
+            "energy",
+            ("https://open.spotify.com/track/" + pl.col("track_id")),
+        )
+        .filter(pl.col("artist_name") == artist)
+        .filter(pl.col("track_name") == track)
+        .limit(1)
+    )
+
+    if not m.is_empty():
+        matches = pl.concat([matches, m])
+
+matches = matches.sort("danceability", descending=True)
+```
+
+A little explination: The above code loops over all the rows in our sample. By providing `named=True`, Polars will also include the colum name in each `row` value. We then look for a single match (via `limit(1)`) in `danceability_tracks` that have both a matching `artist_name` and `track_name`. Additionally, we build a full Spotify open link by prefixing the `track_id` with a base URL. For any matches, we append to the `matches` `DataFrame`. Finally, we sort the results by `danceability`.
+
+Now that we have our `matches`, let's output them to Designer!
+
+```python
+self.provider.write_to_anchor(
+    "Output", pa.Table.from_pandas(matches.to_pandas())
+)
+
+self.provider.io.info(f"{self.name} finished.")
+```
+
+> :information_source: You might notice something a bit odd here: We are constructing a PyArrow table (`pa.Table`) from a Pandas frame converted from Polars. This is due to hhe current version of Polars `to_arrow()` producing somthing not quite right. Perhaps newer versions will no longer require this work around.
```

Comment:
TW review:

> :information_source: You might notice something a bit odd here. We are constructing a PyArrow table (`pa.Table`) from a Pandas frame converted from Polars. This is because the current version of Polars `to_arrow()` produces unexpected results. Perhaps newer versions won't require this workaround.